MayssaraOmar/assignment4/Counting islands.cpp

```diff
@@ -0,0 +1,125 @@
+#include "test.h"
+#include <iostream>
+#include <vector>
+using namespace std;
+
+static void markIsland(
+	const vector<vector<bool>> &grid,
+	int row,
+	int col,
+	vector<vector<bool>> *visited) {
+	// boundary check
+	if (row < 0 || row >= grid.size() || col < 0 || col >= grid[row].size()) {
+		return;
+	}
+	if ((*visited)[row][col] || !grid[row][col]) {
+		return;
+	}
+	(*visited)[row][col] = true;
+	markIsland(grid, row - 1, col, visited); // up
```

Comment:
Maybe having a loop (or two) makes this simpler. Not sure tough, looks good as it is...